kms/main.tf

```diff
@@ -0,0 +1,167 @@
+data "aws_caller_identity" "current" {}
+
+resource "aws_kms_key" "encryption" {
+  description         = var.key_description
+  enable_key_rotation = true
+  policy              = var.key_policy == "" ? data.aws_iam_policy_document.key_policy.json : var.key_policy
+  tags = merge(
+    var.tags,
+    tomap(
+      { "Name" = var.key_name }
+    )
+  )
+}
+
+resource "aws_kms_alias" "encryption" {
+  name          = "alias/${var.key_name}"
+  target_key_id = aws_kms_key.encryption.key_id
+}
+
+module "kms_admin_role" {
+  source             = "../iam_role"
+  assume_role_policy = templatefile("${path.module}/templates/deny_all.json.tpl", {})
+  name               = "${var.key_name}-admin"
```

Comment:
Should we prefix this with `KMS-` so it's clear when looking in the IAM console that this role is a key admin role?

Reply:
I'll do kms because most key names seem to have kebab case names

Reply:
Why did this get reverted?